frontend/catalyst/passes/builtin_passes.py

```diff
@@ -0,0 +1,242 @@
+        raise TypeError(f"A QNode is expected, got the classical function {qnode}")
+
+    clone = copy.copy(qnode)
+    clone.__name__ += "_cancel_inverses"
+
+    @functools.wraps(clone)
+    def wrapper(*args, **kwargs):
+        pass_pipeline = kwargs.pop("pass_pipeline", [])
+        pass_pipeline.append(Pass("remove-chained-self-inverse"))
+        kwargs["pass_pipeline"] = pass_pipeline
+        return clone(*args, **kwargs)
+
+    return wrapper
+
+
+def merge_rotations(qnode):
+    """
+    Specify that the ``-merge-rotations`` MLIR compiler pass
+    for merging roations (peephole) will be applied.
+
+    The full list of supported gates are as follows:
+
+    :class:`qml.RX <pennylane.RX>`,
+    :class:`qml.CRX <pennylane.CRX>`,
+    :class:`qml.RY <pennylane.RY>`,
+    :class:`qml.CRY <pennylane.CRY>`,
+    :class:`qml.RZ <pennylane.RZ>`,
+    :class:`qml.CRZ <pennylane.CRZ>`,
+    :class:`qml.PhaseShift <pennylane.PhaseShift>`,
+    :class:`qml.ControlledPhaseShift <pennylane.ControlledPhaseShift>`,
+    :class:`qml.MultiRZ <pennylane.MultiRZ>`.
+
+
+    .. note::
+
+        Unlike PennyLane :doc:`circuit transformations <introduction/compiling_circuits>`,
+        the QNode itself will not be changed or transformed by applying these
+        decorators.
+
+        As a result, circuit inspection tools such as :func:`~.draw` will continue
+        to display the circuit as written in Python.
+
+        To instead view the optimized circuit, the MLIR must be viewed
+        after the ``"QuantumCompilationPass"`` stage via the
+        :func:`~.get_compilation_stage` function.
+
+    Args:
+        fn (QNode): the QNode to apply the cancel inverses compiler pass to
+
+    Returns:
+        ~.QNode:
+
+    **Example**
+
+    In this example the three :class:`qml.RX <pennylane.RX>` will be merged in a single
+    one with the sum of angles as parameter.
+
+    .. code-block:: python
+
+        from catalyst.debug import get_compilation_stage
+        from catalyst.passes import merge_rotations
+
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qjit(keep_intermediate=True)
+        @merge_rotations
+        @qml.qnode(dev)
+        def circuit(x: float):
+            qml.RX(x, wires=0)
+            qml.RX(0.1, wires=0)
+            qml.RX(x**2, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+    >>> circuit(0.54)
+    Array(0.5965506257017892, dtype=float64)
+    """
+    if not isinstance(qnode, qml.QNode):
+        raise TypeError(f"A QNode is expected, got the classical function {qnode}")
+
+    clone = copy.copy(qnode)
+    clone.__name__ += "_merge_rotations"
+
+    @functools.wraps(clone)
+    def wrapper(*args, **kwargs):
+        pass_pipeline = kwargs.pop("pass_pipeline", [])
+        pass_pipeline.append(Pass("merge-rotations"))
+        kwargs["pass_pipeline"] = pass_pipeline
+        return clone(*args, **kwargs)
+
+    return wrapper
+
+
+def ions_decomposition(qnode):  # pragma: nocover
```

Comment:
@josh146 I didn't export this one to the docs because we've added it for the OQD pipeline and we're not "officially" releasing that stuff yet, is that okay with you or do you want it to appear in the docs anyway? In principle I suppose it could be used independently, what it does is convert `{T, S, Z, Hadamard, RZ, PhaseShift, CNOT}` into `{RX, RY, MS}` gates.

Reply:
@erick-xanadu I will fix the last CI error after answering this question.

Reply:
Alright we decided on making it public.

Reply:
@joeycarter agreed to provide a docstring for this pass since he is stress-testing it right now :)